pandas/core/util/hashing.py

```diff
@@ -164,6 +164,29 @@ def hash_tuples(vals, encoding='utf8', hash_key=None):
     return h
 
 
+def hash_tuple(val, encoding='utf8', hash_key=None):
+    """
+    Hash a single tuple efficiently
+
+    Parameters
+    ----------
+    val : single tuple
+    encoding : string, default 'utf8'
+    hash_key : string key to encode, default to _default_hash_key
+
+    Returns
+    -------
+    hash
+
+    """
+    hashes = (hash_array(np.array([v]), encoding=encoding, hash_key=hash_key,
```

Comment:
> but if you have multiple elements it will not

it will always be a single element I think (it are the individual elements of the single tuple that are put in an array)